pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/utilities/HiveUtilities.java

```diff
@@ -225,15 +245,47 @@ private static boolean verifyModifers(String[] modifiers) {
         }
 
         if (toks.size() == 1) {
-            dbName = HIVE_DEFAULT_DBNAME;
-            tableName = toks.get(0);
+            dbPattern = HIVE_DEFAULT_DBNAME;
+            tablePattern = toks.get(0);
         } else if (toks.size() == 2) {
-            dbName = toks.get(0);
-            tableName = toks.get(1);
+            dbPattern = toks.get(0);
+            tablePattern = toks.get(1);
         } else {
-            throw new IllegalArgumentException("\"" + qualifiedName + "\"" + errorMsg);
+            throw new IllegalArgumentException("\"" + pattern + "\"" + errorMsg);
+        }
+
+        return getTablesFromPattern(client, dbPattern, tablePattern);
+
+    }
+
+    private static List<Metadata.Item> getTablesFromPattern(HiveMetaStoreClient client, String dbPattern, String tablePattern) {
+
+        List<String> databases = null;
+        List<Metadata.Item> itemList = new ArrayList<Metadata.Item>();
+        List<String> tables = new ArrayList<String>();
+
+        if(client == null || (!dbPattern.contains(WILDCARD) && !tablePattern.contains(WILDCARD)) ) {
+            /* This case occurs when the call is invoked as part of the fragmenter api or when metadata is requested for a specific table name */
+            itemList.add(new Metadata.Item(dbPattern, tablePattern));
+            return itemList;
         }
 
-        return new Metadata.Table(dbName, tableName);
+        try {
+            databases = client.getDatabases(dbPattern);
+            if(databases.isEmpty()) {
+                throw new IllegalArgumentException("No database found for the given pattern");
+            }
+            for(String dbName: databases) {
+                for(String tableName: client.getTables(dbName, tablePattern)) {
```

Comment:
can `getTables` accept as parameters both `dbName` pattern and `tablePattern` or does the `dbName` must be a real db?
If it's possible, perhaps we can get the list in one call.